nsq/deflate_socket.py

```diff
@@ -24,6 +24,17 @@ def recv(self, size):
     def read(self, size):
         return self._recv(size, self._socket.read)
 
+    def recv_into(self, buf, nbytes=0):
```

Comment:
why is this method necessary?

Reply:
it's the socket method with tornado-5.x uses instead of `recv()`, for potential efficiency improvement: https://github.com/tornadoweb/tornado/pull/2251/files#diff-46c2c62b2b1964831d9332ba934cb623R1185